tests/handlers/compression/test_xz.py

```diff
@@ -0,0 +1,704 @@
+            id="crc32",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT,
+            0,
+            len(XZ_CRC64_CONTENT),
+            id="crc64",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT,
+            0,
+            len(XZ_SHA256_CONTENT),
+            id="crc_sha256",
+        ),
+        pytest.param(
+            XZ_NONE_CONTENT * 3,
+            0,
+            len(XZ_NONE_CONTENT) * 3,
+            id="crc_none_concat",
+        ),
+        pytest.param(
+            XZ_CRC32_CONTENT * 3,
+            0,
+            len(XZ_CRC32_CONTENT) * 3,
+            id="crc32_concat",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT * 3,
+            0,
+            len(XZ_CRC64_CONTENT) * 3,
+            id="crc64_concat",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT * 3,
+            0,
+            len(XZ_SHA256_CONTENT) * 3,
+            id="sha256_concat",
+        ),
+        pytest.param(
+            XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT,
+            0,
+            len(XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT),
+            id="crc_none_concat_pad",
+        ),
+        pytest.param(
+            XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT,
+            0,
+            len(XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT),
+            id="crc32_concat_pad",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT,
+            0,
+            len(XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT),
+            id="concat_pad",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT,
+            0,
+            len(XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT),
+            id="sha256_concat_pad",
+        ),
+        pytest.param(
+            XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_NONE_CONTENT + NULL_PADDING + XZ_NONE_CONTENT + NULL_PADDING),
+            id="crc_none_concat_pad2",
+        ),
+        pytest.param(
+            XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_CRC32_CONTENT + NULL_PADDING + XZ_CRC32_CONTENT + NULL_PADDING),
+            id="crc32_concat_pad2",
+        ),
+        pytest.param(
+            XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_CRC64_CONTENT + NULL_PADDING + XZ_CRC64_CONTENT + NULL_PADDING),
+            id="crc64_concat_pad2",
+        ),
+        pytest.param(
+            XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT + NULL_PADDING,
+            0,
+            len(XZ_SHA256_CONTENT + NULL_PADDING + XZ_SHA256_CONTENT + NULL_PADDING),
+            id="sha256_concat_pad2",
+        ),
+    ),
+)
+def test_xz_calculate_chunk(content: bytes, start_offset: int, expected_length: int):
```

Comment:
- start_offset is always 0
- expected length is always the length of the content

do we want to specify those or test anything that have other values?

Reply:
Again, mismatch. The start offset is indeed always 0 but the expected length can differ because I have test cases with bad padding.

Reply:
Fixed together.